app/src/main/java/org/mozilla/fenix/settings/account/SignOutFragment.kt

```diff
@@ -63,13 +60,15 @@ class SignOutFragment : BottomSheetDialogFragment() {
         super.onViewCreated(view, savedInstanceState)
 
         view.signOutDisconnect.setOnClickListener {
-            viewLifecycleOwner.lifecycleScope.launch {
+            lifecycleScope.launch {
                 requireComponents
                     .backgroundServices.accountAbnormalities.userRequestedLogout()
                 accountManager.logout()
             }.invokeOnCompletion {
-                if (!findNavController().popBackStack(R.id.settingsFragment, false)) {
-                    dismiss()
+                runIfFragmentIsAttached {
+                    if (!findNavController().popBackStack(R.id.settingsFragment, false)) {
```

Comment:
Where is this `.popBackStack` happening? Do you know why we need this check here? Is this line what is fixing the exception, or is it the `lifecycleScope` on ln 63?